app/components/search/SearchResults/SearchResults.tsx

```diff
@@ -2,52 +2,38 @@ import React, { useState, useEffect } from "react";
 import { Link } from "react-router-dom";
 import ReactMarkdown from "react-markdown";
 import { connectStateResults } from "react-instantsearch/connectors";
+import { Props as GoogleMapProps, Coords } from "google-map-react";
 import { whiteLabel } from "utils";
 import { SearchMap } from "components/search/SearchMap/SearchMap";
 import ResultsPagination from "components/search/Pagination/ResultsPagination";
 import { Texting } from "components/Texting";
 import { ClinicianActions } from "components/ucsf/ClinicianActions/ClinicianActions";
 import { ClientHandouts } from "components/ui/ClientHandoutsModal/ClientHandouts";
+import { SearchHit, transformHits } from "../../../models/SearchHits";
 import { icon } from "../../../assets";
-import { parseAlgoliaSchedule } from "../../../models";
 import styles from "./SearchResults.module.scss";
 
-/**
- * Transform Algolia search hits such that each hit has a recurringSchedule that
- * uses the time helper classes.
- */
-const transformHits = (hits) =>
-  hits.map((hit) => {
-    const inheritedSchedule =
-      hit.schedule && hit.schedule.length
-        ? hit.schedule
-        : hit.resource_schedule;
-    const recurringSchedule =
-      inheritedSchedule && inheritedSchedule.length
-        ? parseAlgoliaSchedule(inheritedSchedule)
-        : null;
-    return { ...hit, recurringSchedule };
-  });
-
-// Todo: setExpandList will be used as part of next stage of multiple location work
-// eslint-disable-next-line no-unused-vars
-const SearchResults = ({ searchResults, expandList, setExpandList }) => {
-  const [centerCoords, setCenterCoords] = useState(null);
-  const [googleMapObject, setMapObject] = useState(null);
+const SearchResults = ({ searchResults, expandList }) => {
+  const [centerCoords] = useState(null);
+  interface MapObjectProps extends GoogleMapProps {
+    setCenter: (coords: Coords) => void;
+  }
+  const [googleMapObject, setMapObject] = useState<MapObjectProps | null>(null);
 
   useEffect(() => {
-    if (centerCoords) {
+    if (centerCoords && googleMapObject) {
       googleMapObject.setCenter(centerCoords);
     }
     document.body.classList.add("searchResultsPage");
+
     return () => {
       document.body.classList.remove("searchResultsPage");
     };
   }, [googleMapObject, centerCoords]);
 
   if (!searchResults) return null;
 
-  const hits = transformHits(searchResults.hits);
+  const hits = transformHits(searchResults.hits as SearchHit[]);
```

Comment:
This might be able to go away if you declare the types of the props of the component.

Reply:
The TS compiler was not happy about the discrepancy between 'Hit<BasicDoc>[]' which is the instantsearch default definition of a hit and our `SearchHit[]`. I tried to extend the `Hit<BasicDoc>` definition when declaring the `SearchHit` interface in the `app/models/SearchHits.ts` but that didn't seem to appease its demands. Anyhow, I was left with doing the following `searchResults.hits as unknown as SearchHit[]` which VSCode suggested as a quick fix...